spec/lib/faye/authentication/http_client_spec.rb

```diff
@@ -14,6 +14,17 @@
       expect(request).to have_been_made
     end
 
+    it 'should not add a signature if the channel is whitelisted' do
+      message = {'channel' => '/foo/bar', 'clientId' => 'http'}
+      message['data'] = 'hello'
+      request = stub_request(:post, "http://www.example.com").with(:body => {:message => JSON.dump(message)}).to_return(:status => 200, :body => "", :headers => {})
+      whitelist = lambda do |channel|
+        channel.start_with?('/foo/')
+      end
+      Faye::Authentication::HTTPClient.publish('http://www.example.com', '/foo/bar', "hello", 'my private key', { whitelist: whitelist })
+      expect(request).to have_been_made
```

Comment:
shouldn''t we check that no signature was added to the message ?

Reply:
It's done through the `stub_request`.

If the spec tries to make a call with the signature, it'll try to make the call because it won't match the stub and the spec with fail because we don't allow real http connections:

```
WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST  http://www.example.com/ with body [...]
```

Reply:
oh right ! my bad I didn't see the `{:message => JSON.dump(message)}` part